repository/repository.go

```diff
@@ -203,3 +203,24 @@ func (r *Repository) Delete(ctx context.Context, id string) error {
 	}
 	return nil
 }
+
+func (r *Repository) Checkout(ctx context.Context, id string) (string, error) {
+	if err := r.exists(ctx, id); err != nil {
+		return "", err
+	}
+
+	branch := "cu-" + id
+
+	// set up remote tracking branch if it's not already there
+	_, err := runGitCommand(ctx, r.userRepoPath, "show-ref", "--verify", "--quiet", fmt.Sprintf("refs/heads/%s", branch))
+	if err != nil {
+		_, err = runGitCommand(ctx, r.userRepoPath, "branch", "--track", branch, fmt.Sprintf("%s/%s", containerUseRemote, id))
+		if err != nil {
+			return "", err
+		}
+	}
+
+	_, err = runGitCommand(ctx, r.userRepoPath, "checkout", branch)
```

Comment:
i think this might actually be wildly overcomplicating things. do we get correct behavior with just plain `git checkout $branch`? maybe with a fetch thrown in? iirc the show-ref and `branch --track` wackiness was only necessary because I wanted the branch setup entirely in the background.